.github/workflows/query-list.yml

```diff
@@ -0,0 +1,39 @@
+name: Build code scanning query list
+
+on:
+  push:
```

Comment:
I agree with your thinking that this should only be pushes to `main`, and perhaps to the RC branches.